test/gns.test.ts

```diff
@@ -708,6 +712,26 @@ describe('GNS', () => {
           'GNS: Cannot update version if not published, or has been deprecated',
         )
       })
+
+      it('should update reserveRatio', async function () {
+        const value = 10
+
+        await gns.setReserveRatio(value)
+
+        const tx = await gns
+          .connect(me.signer)
+          .publishNewVersion(
+            me.address,
+            0,
+            subgraph1.subgraphDeploymentID,
+            subgraph1.versionMetadata,
+          )
+
+        const pool = await gns.nameSignals(me.address, 0)
+        const reserveRatio = pool[3]
```

Comment:
`gns.nameSignals(me.address, 0)` returns an object that lets you access attributes by both position and key:

```
[
  BigNumber { _hex: '0x858ffc3f497c9112', _isBigNumber: true },
  BigNumber { _hex: '0x874391d0dd1fdf42', _isBigNumber: true },
  '0x71a94176484d9e0d2ff853a92e162a0493fafa1823a0841400c054769bd895a1',
  10,
  false,
  BigNumber { _hex: '0x00', _isBigNumber: true },
  vSignal: BigNumber { _hex: '0x858ffc3f497c9112', _isBigNumber: true },
  nSignal: BigNumber { _hex: '0x874391d0dd1fdf42', _isBigNumber: true },
  subgraphDeploymentID: '0x71a94176484d9e0d2ff853a92e162a0493fafa1823a0841400c054769bd895a1',
  reserveRatio: 10,
  disabled: false,
  withdrawableGRT: BigNumber { _hex: '0x00', _isBigNumber: true }
]
```

You should be able to do something like:
`expect(pool.reserveRatio).eq(value)`

Reply:
I was following the pattern used here: https://github.com/graphprotocol/contracts/blob/dev/test/gns.test.ts#L157-L159

Wanted to make sure to edit and mutate as little as possible with this PR

Reply:
I understand. I moved away from using positional arguments on a refactor I did for this PR https://github.com/graphprotocol/contracts/pull/497

We can use the old pattern here and then eventually when audit passes on #497 and if governance decides to move forward with that upgrade we can rebase/merge.